lib/exfile/ecto/cast_content_type.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Exfile.Ecto.CastContentType do
+  alias Ecto.Changeset
+  alias Exfile.Processor.ContentType, as: ContentTypeProcessor
+
+  def cast_content_type(changeset, field) do
+    cast_content_type(changeset, field, "#{field}_content_type")
+  end
+  def cast_content_type(changeset, field, content_type_field) do
+    case Changeset.get_change(changeset, field) do
+      %Exfile.File{} -> perform_cast(changeset, field, content_type_field)
+      _              -> changeset
+    end
+  end
+
+  defp perform_cast(changeset, field, content_type_field) do
+    { :ok, processed_file } = changeset
+      |> Changeset.get_change(field)
+      |> ContentTypeProcessor.call([], [])
+
+    changeset
+    |> Changeset.put_change(field, processed_file)
```

Comment:
I found error here!

We are getting `%Exfile.File{}` struct here, but putting `%Exfile.Localfile{}` to changeset back! This is the reason why things like https://github.com/keichan34/exfile/blob/master/lib/exfile/phoenix/helpers.ex#L76 can't pattern match